plugins/quickhighlight/pluma-quickhighlight-plugin.c

```diff
@@ -0,0 +1,482 @@
+};
+
+enum
+{
+    PROP_0,
+    PROP_VIEW
+};
+
+static void pluma_view_activatable_iface_init (PlumaViewActivatableInterface *iface);
+
+G_DEFINE_DYNAMIC_TYPE_EXTENDED (PlumaQuickHighlightPlugin,
+                                pluma_quick_highlight_plugin,
+                                PEAS_TYPE_EXTENSION_BASE,
+                                0,
+                                G_IMPLEMENT_INTERFACE_DYNAMIC (PLUMA_TYPE_VIEW_ACTIVATABLE,
+                                                               pluma_view_activatable_iface_init)
+                                G_ADD_PRIVATE_DYNAMIC (PlumaQuickHighlightPlugin))
+
+static void pluma_quick_highlight_plugin_notify_buffer_cb (GObject *object, GParamSpec *pspec, gpointer user_data);
+static void pluma_quick_highlight_plugin_mark_set_cb (GtkTextBuffer *textbuffer, GtkTextIter *location, GtkTextMark *mark, gpointer user_data);
+static void pluma_quick_highlight_plugin_delete_range_cb (GtkTextBuffer *textbuffer, GtkTextIter *start, GtkTextIter *end, gpointer user_data);
+static void pluma_quick_highlight_plugin_notify_style_scheme_cb (GObject *object, GParamSpec *pspec, gpointer user_data);
+
+static void
+pluma_quick_highlight_plugin_load_style (PlumaQuickHighlightPlugin *plugin)
+{
+    GtkSourceStyleScheme *style_scheme;
+    GtkSourceStyle *style = NULL;
+
+    g_return_if_fail (PLUMA_IS_QUICK_HIGHLIGHT_PLUGIN (plugin));
+
+    if (plugin->priv->buffer == NULL)
+    {
+        return;
+    }
+
+    pluma_debug (DEBUG_PLUGINS);
+
+    g_clear_object (&plugin->priv->style);
+
+    style_scheme = gtk_source_buffer_get_style_scheme (GTK_SOURCE_BUFFER (plugin->priv->buffer));
+
+    if (style_scheme != NULL)
+    {
+        style = gtk_source_style_scheme_get_style (style_scheme, "quick-highlight-match");
+
+        if (style != NULL)
+        {
+            plugin->priv->style = gtk_source_style_copy (style);
+        }
+    }
+}
+
+static gboolean
+pluma_quick_highlight_plugin_highlight_worker (gpointer user_data)
+{
+    PlumaQuickHighlightPlugin *plugin = PLUMA_QUICK_HIGHLIGHT_PLUGIN (user_data);
+    GtkSourceSearchSettings *search_settings;
+    GtkTextIter start, end;
+    g_autofree gchar *text = NULL;
+
+    g_assert (PLUMA_IS_QUICK_HIGHLIGHT_PLUGIN (plugin));
+
+    plugin->priv->queued_highlight = 0;
+
+    if (!gtk_text_buffer_get_selection_bounds (GTK_TEXT_BUFFER (plugin->priv->buffer), &start, &end))
+    {
+        g_clear_object (&plugin->priv->search_context);
+        return G_SOURCE_REMOVE;
+    }
+
+    if (gtk_text_iter_get_line (&start) != gtk_text_iter_get_line (&end))
+    {
+        g_clear_object (&plugin->priv->search_context);
+        return G_SOURCE_REMOVE;
+    }
+
+    if (plugin->priv->search_context == NULL)
+    {
+        search_settings =
+            g_object_new (GTK_SOURCE_TYPE_SEARCH_SETTINGS,
+                          "at-word-boundaries", FALSE,
+                          "case-sensitive", TRUE,
+                          "regex-enabled", FALSE,
```

Comment:
I would love for these to be settings exposed either through the plugin Preferences UI or some gsettings

Reply:
I personally do not need this to be configurable. But if someone likes to do it, a PR would be nice of course.